zerver/actions/message_edit.py

```diff
@@ -930,6 +930,98 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
     return len(changed_messages)
 
 
+def check_time_limit_for_change_all_propagate_mode(
+    message: Message,
+    user_profile: UserProfile,
+    topic_name: Optional[str] = None,
+    stream_id: Optional[int] = None,
+) -> None:
+    realm = user_profile.realm
+    if (
+        realm.move_messages_within_stream_limit_seconds is None
+        and realm.move_messages_between_streams_limit_seconds is None
+    ):
+        # There is no time limit for editing both stream and topics, so we need
+        # not check further and moving message is allowed.
+        return
+
+    message_move_limit_buffer = 20
+
+    topic_edit_deadline_seconds = None
+    if topic_name is not None and realm.move_messages_within_stream_limit_seconds is not None:
+        # We set topic_edit_deadline_seconds only if topic is actually changed and there is
+        # some time limit to edit topic.
+        topic_edit_deadline_seconds = (
+            realm.move_messages_within_stream_limit_seconds + message_move_limit_buffer
+        )
+
+    stream_edit_deadline_seconds = None
+    if stream_id is not None and realm.move_messages_between_streams_limit_seconds is not None:
+        # We set stream_edit_deadline_seconds only if stream is actually changed and there is
+        # some time limit to edit stream.
+        stream_edit_deadline_seconds = (
+            realm.move_messages_between_streams_limit_seconds + message_move_limit_buffer
+        )
+
+    # If the code reaches here, then there is a time limit to edit at least one of topic or stream.
+    assert topic_edit_deadline_seconds is not None or stream_edit_deadline_seconds is not None
```

Comment:
This assertion is I wrong, in that you could have a time limit only on moving messages between streams, but this specific request is only to move the topic within the stream. I fixed this by just making the `return` case an `else` clause a bit further down, which also feels a lot more readable.

As a side note, I line-wrapped a bunch of your comments; generally we want to line-wrap comments to ~72 characters, unless they contain a URL or something that provides a reason not to do so.

Reply:
Thanks for catching that.